sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageStreamSource.java

```diff
@@ -146,8 +164,19 @@ public String toString() {
     return readStream.toString();
   }
 
+  @Override
+  public long getMaxEndOffset(PipelineOptions options) throws Exception {
+    return getEndOffset();
+  }
+
+  @Override
+  public OffsetBasedSource<T> createSourceForSubrange(long start, long end) {
+    return this;
```

Comment:
This is incorrect since the runner may try to split the source at various ranges and run work items for each split which will result in duplicate data if this change is committed.